cookbooks/planet/templates/default/planetdump.erb

```diff
@@ -74,6 +74,83 @@ time nice -n 19 /opt/planet-dump-ng/planet-dump-ng \
      -x "planet-${date}.osm.bz2" -X "history-${date}.osm.bz2" \
      -p "planet-${date}.osm.pbf" -P "history-${date}.osm.pbf"
 
+# Function to create bittorrent files
+function mk_torrent {
+  type="$1"
+  format="$2"
+  web_dir="$3"
+  name="${type}-${date}.osm.${format}"
+  web_path="${web_dir}/${name}"
+
+  mktorrent -l 22 ${name} \
+     -a udp://tracker.opentrackr.org:1337 \
+     -a udp://tracker.datacenterlight.ch:6969/announce,http://tracker.datacenterlight.ch:6969/announce \
+     -a udp://tracker.torrent.eu.org:451 \
+     -a udp://tracker-udp.gbitt.info:80/announce,http://tracker.gbitt.info/announce,https://tracker.gbitt.info/announce \
+     -a http://retracker.local/announce \
+     -w https://ftp5.gwdg.de/pub/misc/openstreetmap/planet.openstreetmap.org/${web_path} \
+     -w https://ftpmirror.your.org/pub/openstreetmap/${web_path} \
+     -w https://mirror.init7.net/openstreetmap/${web_path} \
+     -w https://planet.openstreetmap.org/${web_path} \
+     -w https://free.nchc.org.tw/osm.planet/${web_path} \
+     -w https://ftp.fau.de/osm-planet/${web_path} \
+     -w https://ftp.spline.de/pub/openstreetmap/${web_path} \
+     -w https://osm.openarchive.site/${name} \
+     -w https://downloads.opencagedata.com/planet/${name} \
```

Comment:
Can we add https://ftp.osuosl.org/pub/openstreetmap/?

Reply:
## about XML
I'd agree with handcrafted XML problems if we had more variable input, but the input for generating XML is **quite fixed** here. There are only 3 things that ever change (the rest is extracted to variables for readability and removing redundancy):
  - 6-digit numeric part of name (eg. `planet-201109` becomes `planet-201116`), as created elsewhere in code.
  - purely integer size of file in bytes (as returned by `find -printf "%s"`)
  - date in RFC 5322 format as returned by `date -R` (which ignores locale, and contains only US-ASCII aphanumerics, spaces, `:` for hours/minutes/seconds separation, and `+` or `-` to indicate timezone).

None of the above is problematic for XML generated, nor needs escaping. I've also validated the resulting XML at https://validator.w3.org/feed/ and it is valid. Since input doesn't otherwise change, I think it should be quite good. 
_(But if it is requirement of operations team to separate it in some other programing language which uses proper XML module, I can redo it in perl. But I hope my explanation clears this issue well enough?)_

## about extra webseed
Unfortunately, not at the moment, as mentioned briefly in "webseed selection" section above. https://ftp.osuosl.org/pub/openstreetmap/ provides `.pbf` only with name `planet-latest.osm.pbf` and we need it in format `planet-201109.osm.pbf`. If they can change their system to have latest file in format `planet-YYMMDD.osm.pbf` (and perhaps add `planet-latest.osm.pbf` as symlink if they want it) then they can be added too!
This is important due to race condition when `planet.openstreetmap.org` had published the torrent, but `osuosl.org` didn't yet finished fetching the new planet, then `osuosl.org` would serve blocks from wrong (old) `latest` file, which would poison the swarm with invalid blocks. 
Using `planet-YYMMDD.osm.pbf` as filename fixes that issue, as clients will get `404` if mirror is not yet synced, which they will handle correctly (by trying another peer / webseed)